seq_tools/validation/c160_file_r1_r2_check.py

```diff
@@ -0,0 +1,118 @@
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+        Linda Xiang <linda.xiang@oicr.on.ca>
+"""
+
+
+from base_checker import BaseChecker
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata):
+        super().__init__(ctx, metadata, __name__)
+
+    def check(self):
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(message)
+            self.message = message
+            self.status = 'INVALID'
+            return
+ 
+        if not self.metadata.get('files'):
+            message = "Missing 'files' section in the metadata JSON"
+            self.logger.info(message)
+            self.message = message
+            self.status = 'INVALID'
+            return
+
+        fns = set()
+        for rg in self.metadata.get('read_groups'):
+            if 'is_paired_end' not in rg:
+                message = "Required field 'is_paired_end' is not found in metadata JSON"
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+
+            if not isinstance(rg['is_paired_end'], bool):
+                message = "Required field 'is_paired_end' should be Boolean type."
+                self.logger.info(message)
+                self.message = message
+                self.status = 'INVALID'
+                return
+            
+            if rg['is_paired_end']:
+                if not 'file_r1' in rg or not rg['file_r1']:
+                    message = "Required field 'file_r1' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if not 'file_r2' in rg or not rg['file_r2']:
+                    message = "Required field 'file_r2' is not found in metadata JSON for paired end sequencing reads."
+                    self.logger.info(message)
+                    self.message = message
+                    self.status = 'INVALID'
+                    return
+                if rg['file_r1'].endswith('.bam') and not rg['file_r1'] == rg['file_r2']:
+                    message = "Fields 'file_r1' and 'file_r2' should be the same for paired end BAM sequencing reads."
```

Comment:
consider adding `submitter_read_group_id` in the message, similar to the comment above